python_modules/dagster/dagster/_utils/error.py

```diff
@@ -94,6 +97,106 @@ def _should_redact_user_code_error() -> bool:
 )
 
 
+redacted_exception_id_to_user_facing_identifier: dict[int, str] = {}
```

Comment:
https://docs.python.org/3/library/weakref.html#weakref.WeakKeyDictionary i think might be better